lib/bolt/pal.rb

```diff
@@ -52,12 +52,21 @@ def initialize(modulepath, hiera_config, resource_types, max_compiles = Etc.npro
       @apply_settings = apply_settings
       @max_compiles = max_compiles
       @resource_types = resource_types
+      @project = project
 
       @logger = Logging.logger[self]
       if modulepath && !modulepath.empty?
         @logger.info("Loading modules from #{@modulepath.join(File::PATH_SEPARATOR)}")
       end
 
+      # If the project name is the same as one of the builtin modules raise a warning
+      bolt_mods = Dir.children(BOLTLIB_PATH)
+      if @project && bolt_mods.include?(@project&.name)
```

Comment:
```suggestion
      if @project && bolt_mods.include?(@project.name)
```

With the change to `Bolt::Project#name`, this will always return a value and also pick up directory names that conflict with core modules.